### PR TITLE
feat: Add Minikube documentation

### PR DESCRIPTION
**Pull request by afbjorklund**
### What does this PR do?

This is the documentation for the minikube extension. It is rather similar to the "kind" extension.

Both Kubernetes extensions share a lot of concepts, even though the commands differ slightly.

### What issues does this PR fix or reference?

* #2610

### How to test this PR?

* #2646

Note for reviewer: this "minikube" documentation should be compared with the existing "kind" documentation:

* https://podman-desktop.io/docs/kubernetes/kind

### CODE REVIEW

**Comment by benoitf** [comment]
cc @themr0c

**Comment by benoitf** [comment]
would be nice if we could merge some items to maybe avoid duplication but can be done as a next thing once it'll be merged. 
Need to have the extension in its own repository before merging here

**Comment by afbjorklund** [comment]
Now available:

`ghcr.io/containers/podman-desktop-extension-minikube`

File: website/docs/kubernetes/minikube/configuring-podman-for-minikube-on-windows.md [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+tags: [migrating-to-kubernetes, minikube]
+---
+
+# Configuring Podman for Minikube on Windows Subsystem for Linux (WSL)
+
+When you create a Podman machine, Podman creates the machine in rootless mode.
+
+With a Podman machine running on WSL, Minikube:
+
+- Requires the rootful machine.
```
Comment (cdrage) [inline website/docs/kubernetes/minikube/configuring-podman-for-minikube-on-windows.md]:
```suggestion
- Requires a rootful machine.
```

File: website/docs/kubernetes/minikube/pushing-an-image-to-minikube.md [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+
+#### Verification
+
+Minikube enables you to list loaded images, using:
+
+```command
+$ minikube image list
+```
+
+You can also create a Pod that uses the loaded image.
```
Comment (cdrage) [inline website/docs/kubernetes/minikube/pushing-an-image-to-minikube.md]:
```suggestion
You can also create a Pod that uses the loaded image:
```

File: website/docs/kubernetes/minikube/restarting-your-minikube-cluster.md [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+1. Open **Containers**.
+1. Find the Minikube cluster that restarted.
+1. The cluster **Age** is consistent with the restart time.
+1. Open **Pods**.
+1. Find the pods that are running on your Minikube cluster.
+
+#### Workaround
+
+Minikube has no command to restart a cluster.
+Therefore, Podman Desktop stops the Minikube cluster, starts it again, and hopes for the best.
```
Comment (cdrage) [inline website/docs/kubernetes/minikube/restarting-your-minikube-cluster.md]: Should we remove the "hopes for the best" part?

Comment (cdrage) [reply website/docs/kubernetes/minikube/restarting-your-minikube-cluster.md]: The three sentences can be "Minikube has no command to restart a cluster, so you can: blahblah"

**Review by cdrage — CHANGES_REQUESTED** [review]
Thanks for the awesome documentation! I only have a few comments.

Also, minikube on all official documentation (github, https://minikube.sigs.k8s.io/docs/) is stylized as lowercase `minikube` not `Minikube`. 

Unsure of the english on it, but I think minikube should be lowercase on all references? Just a small nitpick. I just remember docs teams sometimes being very specific on whether or not something is capitalized. Like how crc is not Crc and minishift is Minishift, and then microshift is MicroShift. Just a thought

**Comment by afbjorklund** [comment]
I think it used to be a thing also in minikube, but I'm not sure it actually matters to anyone anymore...

Except for in the logo, perhaps ?

minikube logo with name

Trying to spell things like CRC or CRI-O seem to always end up failing anyway - so might as well use English.

**Comment by afbjorklund** [comment]
> Thanks for the awesome documentation! I only have a few comments.

Note that since most of this Kubernetes documentation has been duplicated from the Kind-specific section, most of the things being addressed applies also to _that_ original documentation. Maybe they could be merged, in the future?

 
Example:
```diff
-# Configuring Podman for Kind on Windows Subsystem for Linux (WSL)
+# Configuring Podman for Minikube on Windows Subsystem for Linux (WSL)
 
 When you create a Podman machine, Podman creates the machine in rootless mode.
 
-With a Podman machine running on WSL, Kind:
+With a Podman machine running on WSL, Minikube:
 
 - Requires the rootful machine.
 ```

**Review by benoitf — APPROVED** [review]

**Comment by afbjorklund** [comment]
@benoitf : thanks, will you add some instructions on how to install the minikube extension ?

https://podman-desktop.io/docs/extensions/install

**Comment by benoitf** [comment]
@afbjorklund we could replace the placeholder on https://podman-desktop.io/docs/extensions/install being quay.io/fbenoit/my-first-extension by the minikube extension
but we're still missing in extension page the list of ALL extensions, not only featured extensions

**Comment by afbjorklund** [comment]
I was thinking there needs to be a link from the minikube docs, until the featured (or catalog) is in place

**Comment by benoitf** [comment]
@afbjorklund yes it does not prevent to have links

**Pull request merged**